lib/iris/_concatenate.py

```diff
@@ -464,6 +465,12 @@ def match(self, other, error_on_mismatch):
             msgs.append(msg_template.format('Datatypes', '',
                                             self.data_type, other.data_type))
 
+        # Check _cell_measures_and_dims
+        if self._cell_measures_and_dims != other._cell_measures_and_dims:
```

Comment:
@marqh Update the doc-string to mention the cell measures are now considered part of the `_CubeSignature` matching criteria.